R/html-tweak.R

```diff
@@ -129,6 +129,50 @@ tweak_footnotes <- function(html) {
   xml2::xml_remove(container)
 }
 
+tweak_sourcecode_divs <- function(html) {
+  divs <- xml2::xml_find_all(html, ".//div[contains(@class, 'sourceCode')]")
+  purrr::walk(divs, tweak_sourcecode_div)
+}
+
+tweak_sourcecode_div <- function(html) {
+  pre_class <- xml2::xml_attr(xml2::xml_find_first(html, "pre"), "class")
+  is_r <- grepl("downlit", pre_class) || grepl("sourceCode [rR]", pre_class)
+  if (!is_r) {
+    text <- xml2::xml_text(html)
+  } else {
+    copy <- copy_html(html)
+    # remove output
+    xml2::xml_remove(xml2::xml_find_all(copy, "//span[@class='r-out co']"))
```

Comment:
Should pkgdown (or downlit) also add this class to some comments based on `knitr::opts_chunk$get()$comment`? But how would one get this value once the document is knit?

Reply:
Maybe it'd be easier to remove all comments. Alternatively one would need more complexity e.g. knitting all documents with a hook (that seems interesting for bs4_book actually); but the README.Rmd would escape this treatment...

Reply:
I think we should just remove all lines starting with `#>`. It's going to be too complicated to support all the various knitr customisations that folks might have made.

Reply:
So no knitr hook? :slightly_smiling_face: (I agree it's simpler to assume `#>`!)

Reply:
Yup